packages/kbn-pm/src/utils/bazel/setup_remote_cache.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import dedent from 'dedent';
+import { existsSync, writeFileSync } from 'fs';
+import { resolve } from 'path';
+import { spawn } from '../child_process';
+import { log } from '../log';
+
+async function isVaultAvailable() {
+  try {
+    await spawn('vault', ['--version'], { stdio: 'pipe' });
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+export async function setupRemoteCache(repoRootPath: string) {
+  log.debug(`[bazel_tools] setting up remote cache settings if necessary`);
+
+  const settingsPath = resolve(repoRootPath, '.bazelrc.cache');
+
+  if (existsSync(settingsPath)) {
+    log.debug(`[bazel_tools] remote cache settings already exist, skipping`);
+    return;
+  }
+
+  if (!(await isVaultAvailable())) {
+    log.info('[bazel_tools] vault is not available, unable to setup remote cache settings.');
+    log.info('[bazel_tools] building packages will work, but will be slower in many cases.');
+    log.info('[bazel_tools] to setup vault, if you are an Elastic employee, visit:');
+    log.info('[bazel_tools] https://github.com/elastic/infra/tree/master/docs/vault#github-auth');
+    return;
+  }
+
+  let apiKey = '';
+
+  try {
+    const { stdout } = await spawn(
+      'vault',
+      ['read', '-field=readonly-key', 'secret/kibana-issues/dev/bazel-remote-cache'],
+      {
+        stdio: 'pipe',
+      }
+    );
+    apiKey = stdout.trim();
+  } catch (ex: unknown) {
+    log.info(
+      '[bazel_tools] unable to read bazel remote cache key from vault, are you authenticated?'
+    );
+    log.info('[bazel_tools] building packages will work, but will be slower in many cases.');
+    log.info('[bazel_tools] to setup vault, if you are an Elastic employee, visit:');
+    log.info('[bazel_tools] https://github.com/elastic/infra/tree/master/docs/vault#github-auth');
```

Comment:
I don't think we're supposed to link to private repositories, but I will double-check. In the case which we can't, we can copy the relevant parts of that doc into a new docs page and link to that.

Reply:
We could also limit this message to internal users by checking the email like [here](https://github.com/elastic/kibana/blob/main/packages/kbn-dev-utils/src/ci_stats_reporter/ci_stats_reporter.ts#L140).